subsys/net/lib/sockets/socket_dispatcher.c

```diff
@@ -318,6 +328,26 @@ static int sock_dispatch_setsockopt_vmeth(void *obj, int level, int optname,
 			/* Native interface - use native socket implementation. */
 			fd = sock_dispatch_native(obj);
 		}
+	} else if ((level == SOL_TLS) && (optname == TLS_NATIVE)) {
+		const int *tls_native = (int *)optval;
+		struct dispatcher_context *ctx = (struct dispatcher_context *)obj;
```

Comment:
Implicit casting should work without any issues here, as we only try to cast from/to `void *`. IMO it is rather harmful in that case to use explicit casting, as compiler warnings will be suppressed in the case when we change `sock_dispatch_setsockopt_vmeth()` argument types and forgot to update those explicit casts.

Reply:
Dropped the casting.